src/Root.cpp

```diff
@@ -607,7 +607,7 @@ bool cRoot::FindAndDoWithPlayer(const AString & a_PlayerName, cPlayerListCallbac
 			size_t Rating = RateCompareString (m_PlayerName, a_pPlayer->GetName());
 			if ((Rating > 0) && (Rating >= m_BestRating))
 			{
-				m_BestMatch = a_pPlayer;
+				m_BestMatch = a_pPlayer->GetName();
```

Comment:
If there are two players with the same name it will fail the if condition on line 639 and the callback just won't be called. This might be a bug in the existing implementation as the documentation implies that the callback is called on all players in that scenario, where as the current and proposed implementations will just fail.